app/controllers/api/requests_controller.rb

```diff
@@ -66,6 +66,10 @@ def requests_search_conditions
       {:requester => User.current_user}
     end
 
+    def fetch_requests_href_slug(resource)
+      "requests/#{resource.id}"
```

Comment:
I still feel that it's not up to the controller to change the resource's href slug on the fly - is this open for discussion?